cmd/helm/upgrade.go

```diff
@@ -142,7 +145,37 @@ func newUpgradeCmd(cfg *action.Configuration, out io.Writer) *cobra.Command {
 			if err != nil {
 				return err
 			}
+
+			// Only check dependencies if there are any
 			if req := ch.Metadata.Dependencies; req != nil {
+				// Update all dependencies if DependencyUpdate is true
+				if client.DependencyUpdate {
```

Comment:
Since this code is identical to the code in `helm install`, can you please refactor this out into a separate function? That way the code can remain [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself).

Reply:
@bacongobbler Thanks for the Review. I tried my best with my newbie go knowledge.. :-). Note: I did not include the client in the function, since the Struct differs between the Install and Upgrade Class and that would have required some changes I wasn't confident doing.